### PR TITLE
Python 3 Support

### PR DESCRIPTION
**Pull request by ahurka**
Compatibility for Python 3.5 added without breaking Python 2.7. Travis and Appveyor updated to run tests for Python 3.5, and some infrastructure added for handling version-specific changes in dependencies.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ language: python
 
 python:
   - "2.7"
```
Comment (tomkralidis) [inline .travis.yml]: Let's remove 2.7 here?

File: .travis.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ language: python
...
-#  - "3.4"
```
Comment (tomkralidis) [inline .travis.yml:5]: Remove 3.4 as well.

File: appveyor.yml [hunk 3]
```diff
@@ -5,6 +5,10 @@ environment:
       PYTHON_VERSION: "2.7"
```
Comment (tomkralidis) [inline appveyor.yml]: Remove 2.7 here as well.

File: requirements.txt [hunk 4]
```diff
@@ -1,2 +1,2 @@
-unicodecsv
+unicodecsv  # Only required with Python 2
```
Comment (tomkralidis) [inline requirements.txt]: here too.

File: tests/run_tests.py [hunk 5]
```diff
@@ -43,12 +43,17 @@
...
-import csv
 import io
 import unittest
-from StringIO import StringIO
-from woudc_extcsv import (dump, dumps, load, loads, Reader,
-                          WOUDCExtCSVReaderError, Writer)
+from woudc_extcsv import (dump, dumps, load, loads, kw_restrict,
+                          Reader, WOUDCExtCSVReaderError, Writer)
+
+try:
```
Comment (tomkralidis) [inline tests/run_tests.py]: Remove try/except to support only Python 3.

File: tests/run_tests.py [hunk 6]
```diff
@@ -78,8 +83,15 @@ def get_file_string(file_path):
     """helper function, to open test file and return
        unicode string of file contents
     """
-    with io.open(file_path, 'rb') as f:
-        return f.read()
+
+    try:
```
Comment (tomkralidis) [inline tests/run_tests.py:83]: Remove try/except to support only Python 3.

File: woudc_extcsv/__init__.py [hunk 7]
```diff
@@ -43,20 +43,36 @@
 #
 # =================================================================
 
-import unicodecsv as csv
+from __future__ import print_function
```
Comment (tomkralidis) [inline woudc_extcsv/__init__.py]: What do we need this for? Can we remove in support of Python 3 only?

File: woudc_extcsv/__init__.py [hunk 8]
```diff
@@ -43,20 +43,36 @@
...
-from StringIO import StringIO
+
 from collections import OrderedDict
 from pywoudc import WoudcClient
 
-__version__ = '0.2.2'
 
 LOGGER = logging.getLogger(__name__)
 
+try:
```
Comment (tomkralidis) [inline woudc_extcsv/__init__.py]: Remove try/except to support only Python 3.

**Review by tomkralidis — CHANGES_REQUESTED** [review]

**Comment by ahurka** [comment]
README.md is still in need of updates.

File: woudc_extcsv/__init__.py [hunk 9]
```diff
@@ -115,8 +115,8 @@ def __init__(self, content, parse_tables=False, encoding='utf-8'):
                 body.replace('%', ',')
             try:
                 s = StringIO(body)
-                c = csv.reader(s, encoding=encoding)
-            except Exception as err:
+                c = csv.reader(s)
+            except Exception:
```
Comment (tomkralidis) [inline woudc_extcsv/__init__.py]: Can you add back in and log the error?
```python
except Exception as err:
    LOGGER.warning(err)
    c = csv.reader(s)
```

Comment (ahurka) [reply woudc_extcsv/__init__.py]: Do you want this to replace the existing line? Most of the other try-catches use this method of logging, should they all be changed to use the logger?

`self.errors.append(_violation_lookup(0))`

**Pull request merged**